homeassistant/components/august/__init__.py

```diff
@@ -317,11 +278,11 @@ def lock_has_doorsense(self, device_id):
         await self._async_update_locks_detail()
         return self._lock_detail_by_id[device_id]
 
-    def get_lock_name(self, device_id):
-        """Return lock name as August has it stored."""
-        for lock in self._locks:
-            if lock.device_id == device_id:
-                return lock.device_name
+    def get_device_name(self, device_id):
+        """Return doorbell or lock name as August has it stored."""
+        for device in self._locks + self._doorbells:
```

Comment:
If you use `chain` from itertools you don't need to create a new list each time this method is called.

Reply:
I don't know how often you access `self._locks` and `self._doorbells`, but might be more useful to store it as a dictionary keyed by device ID? That way you can still iterate it as a list using `.values()` and you can do instant lookups.

Reply:
Switched to chain.  Thanks for the tip, its been so long since I wrote any python that I'm rusty on the utility/lib names.

Iteration of the base self._locks/self._doorbells is rare.  The only other place I think they are used is in setup.  The detail lookup is already a dict which is what is used 99% of the time.  We just can't use it in this instance because it might be None (likely so since if we are looking up the name an api operation failed)